gpu_bdb/bdb_tools/q08_utils.py

```diff
@@ -91,8 +101,12 @@ def get_session_id(df):
     df["session_change_flag"] = df["review_flag"] | df["user_change_flag"]
 
     df = df.reset_index(drop=True)
-    df["t_index"] = cp.arange(start=0, stop=len(df), dtype=np.int32)
-
+    
+    if isinstance(df, cudf.DataFrame):
+        df["t_index"] = cp.arange(start=0, stop=len(df), dtype=np.int32)
```

Comment:
You can you `like`  argument/nep 18 to prevent branching here.

Reply:
What do you mean by this? @VibhuJawa

Reply:
We can remove this branching by using below logic : 
```python
df["t_index"] = np.arange(start=0,   stop=len(df),   dtype=np.int32,  like=df["wcs_user_sk"].values)
```

If `df["wcs_user_sk"]` is `cuDF` series, `df["wcs_user_sk"].values` is a cupy array and using like will mean that 
` np.arange(start=0,   stop=len(df),   dtype=np.int32,  like=df["wcs_user_sk"].values)` will be of type `cupy.ndarray`


If `df["wcs_user_sk"]` is `pandas` series,  `df["wcs_user_sk"].values` is a numpy array and using like will mean that 
` np.arange(start=0,   stop=len(df),   dtype=np.int32,  like=df["wcs_user_sk"].values)` will be of type `np.ndarray`